src/Uno.UWP/UI/Colors.cs

```diff
@@ -27,23 +27,21 @@ static partial class Colors
 		/// <returns></returns>
 		public static Color Parse(string colorCode)
 		{
-			if (colorCode?.StartsWith("#", StringComparison.Ordinal) ?? false)
+			if (!string.IsNullOrWhiteSpace(colorCode))
 			{
-				return FromARGB(colorCode);
-			}
-			else
-			{
-				if (!string.IsNullOrWhiteSpace(colorCode))
+				if (colorCode[0] == '#')
 				{
-					Color color;
-
-					if (!_colorMap.TryGetValue(colorCode, out color))
+					return FromARGB(colorCode);
+				}
+				else
+				{
+					if (!_colorMap.TryGetValue(colorCode, out var color))
 					{
 						var property = typeof(Colors).GetProperty(colorCode);
 
 						if (property != null)
 						{
-							_colorMap[colorCode] = color = (Color)property.GetValue(null);
+							_colorMap[colorCode] = color = (Color)property.GetValue(null)!;
```

Comment:
Can we get improved performance by hard-coding the dictionary and avoid reflection?

We can add a reflection-based unit test to ensure the dictionary is correct.

Reply:
Yes, but that's for a different PR :)

The current code seems to have a potential issue with the linker, e.g. a property could be removed (not directly referenced) and that would change how a color is accepted or not. That's tricky to debug... so it might be better to refactor the properties code too.

Reply:
Good catch. Indeed the current code will definitely fail if the linker removes members, and definitely, using an hardcoded table will help performance for bound string lookups.